lib/Controller/UsersController.php

```diff
@@ -199,4 +165,41 @@ public function get(string $userId): DataResponse {
 
 		return new DataResponse($guests);
 	}
+
+	public function transfer(string $email, string $userId, string $password): DataResponse {
+		$user = $this->userManager->get($email);
+		if (!($user instanceof IUser)) {
+			throw new OCSNotFoundException();
+		}
+
+		if (!$this->guestManager->isGuest($user)) {
+			return new DataResponse([], Http::STATUS_CONFLICT);
+		}
+
+		$newUser = $this->userManager->createUser(
+			$userId,
+			$password,
+		);
+
+		if (!($newUser instanceof IUser)) {
+			throw new OCSException('Failed to create new user');
+		}
+
+		$newUser->setSystemEMailAddress($email);
+
+		try {
+			$this->guestManager->transfer($user, $newUser);
+			$result = $user->delete();
+			if (!$result) {
+				$this->logger->error('Failed to delete user', [ 'userId' => $user->getUID() ]);
+			}
+		} catch (\Throwable $th) {
+			$result = $newUser->delete(); // Rollback created user
+			if (!$result) {
+				$this->logger->error('Failed to delete user', [ 'userId' => $newUser->getUID() ]);
+			}
+			throw new OCSException('Failed to transfer guest');
+		}
+		return new DataResponse([], Http::STATUS_CREATED);
```

Comment:
Maybe return the created user so it can be used directly in the frontend?

Reply:
Not needed with current implementation